emartech/3rd-party-integrations-magento-M1
# Version 0.2.4 Updates

Extension Version: 0.2.4

Installation / Upgrade Instructions:
-----------------------------------
1. Delete all the old extension files (below)
    a. js/emarsys
    b. app/code/community/Emarsys
    c. app/etc/modules/Emarsys_Suite.xml
    d. app/etc/modules/Emarsys_Suite2.xml
    e. app/etc/modules/Emarsys_Suite2email.xml
    f. app/etc/modules/Emarsys_Webextend.xml
    g. app/design/frontend/base/default/layout/webextend.xml
    h. app/design/frontend/base/default/template/emarsys
    i. app/design/adminhtml/default/default/layout/emarsys.xml
    j. app/design/adminhtml/default/default/layout/webextend.xml
    k. app/design/adminhtml/default/default/template/emarsys
2. Unzip the provided new extension package and copy all the files from "emarsys-m1-ver0.2.3" directory to the Magento root directory
3. Flush / Refresh All Magento Cache
4. Logout from Magento admin and login back


Enhancements:
-------------
A. Product Full Catalog Export:
    1. Use API Post Method to export the product catalog to Emarsys
    2. Provide a configuration for the Below fields under Catalog export settings
	a. Export Method (API POST / SFTP) [By Default API POST]
	b. Merchant ID
	c. Bearer Token
	d. Test Connection [Button]
    3. Make Sure the API Post works aside of FTP feature
    4. When the "Test Conncetion" button clicked push a CSV with only headers & one sample product and run the test

B. Sales Data (SI) Export:
    1. Use API Post Method to export the Orders to Emarsys
    2. Configuration for the Below fields under Catalog export settings
	a. Export Method (API POST / SFTP) [By Default API POST]
	b. Merchant ID
	c. Bearer Token
	d. Maximum Number of records per export
	e. Test Connection
    3. Make Sure the API Post works aside of FTP feature
    4. When the "Test Conncetion" button clicked push a CSV with only headers & one sample order and run the test

C. Clear the chunk of Customer records from Queue as soon as synced to Emarsys

D. Clear the chunk of Subscriber records from Queue as soon as synced to Emarsys

E. Update the Label of "Export Subscribers" (Contact Synchronization) to "Sync Subscription Values"

F. Notification Feature:
    1. Latest extension version [get the tag_name (Version Number)] from https://api.github.com/repos/emartech/3rd-party-integrations-magento-M1/releases/latest
    2. Display the below information on the About section (System => Configuration => Emarsys Connect => About Emarsys Extension)
	a. Installed extension version
	b. Latest extension version
	c. Read More link to Emarsys Releases
    3. When the About section ( System => Configuration => Emarsys Connect => About Emarsys Extension ) clicked, 
	a. Add a notification to Magento admin if the new release available
	b. Mark as unread if there is existing notification is not older than a week
    4. Magento Community & Enterprise Edition Support

================================================================================================
